Mathlib/Topology/MetricSpace/Katetov.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2024 Luigi Massacci. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luigi Massacci
+-/
+
+import Mathlib.Topology.MetricSpace.Basic
+import Mathlib.Topology.UniformSpace.Cauchy
+
+/-!
+# Katetov Maps
+
+In this file we define Katetov maps (i.e. one point extensions of a metric) and establish
+basic properties of their space. We embed a metric space in its space of Katetov maps via
+the Kuratowski embedding.
+
+## Notation
+
+ - `E(X)` is the type of Katetov maps on `X`.
+
+## References
+
+* [melleray_urysohn_2008]
+-/
+
+variable {α : Type _} [MetricSpace α]
+
+/-- A real valued function from a metric space is katetov if
+  it satisfies the following inequalities: -/
+@[mk_iff]
+structure IsKatetov (f : α → ℝ) : Prop where
+  /-- Proposition that `f` is 1-Lipschitz -/
+  abs_sub_le_dist : ∀ x y, |f x - f y| ≤ dist x y
+  /-- Second defining inequality of a Katetov map  -/
+  dist_le_add : ∀ x y, dist x y ≤ f x + f y
+
+/-- The type of Katetov maps from `α` -/
+structure KatetovMap (α : Type*) [MetricSpace α] where
+  /-- The function `α → ℝ` -/
+  protected toFun : α → ℝ
+  /-- Proposition that `toFun` is a Katetov map -/
+  protected IsKatetovtoFun : IsKatetov toFun
+
+/-- The type of Katetov maps from `α`. -/
+notation "E(" α ")" => KatetovMap α
```

Comment:
Please make this notation scoped, so that it can be selectively enabled in other files.